### PR TITLE
The Xanu Frigate

### PR DESCRIPTION
**Pull request by nauticall**
WIP.

Adds the All-Xanu Spacefleet Shrike-class frigate, a comparatively colossal warship in service with Xanu Prime's navy. Built for combat, it has a myriad of features designed to make it stand toe-to-toe against most enemy combatants.

It features:

- Separate navigation bridge and CIC, a la the Horizon's command bunker. The CIC has its own air and power supply.
- Redundant thruster nacelles; even if one nacelle is destroyed, the ship can continue operating on a single one.
- Two fusion reactors. The power required by this thing is just that huge.
- Thiccccc armor plating and whipple shields made of grating to protect against meteor and dust strikes.
- A Longbow and Grauwolf.
- A fightercraft shuttle, with a Francisca.
- A boarding shuttle.
- Fully-stocked medbay with operating room.
- A brig.

The ghostrole capacity for this is 7 (6 crew + 1 captain), but has capacity for all the way up to 15 (14 crew + 1 captain) during events.

The intended purpose of this frigate is twofold: a reduced-power "patrol" configuration for light skirmishes, and a full-armament "battle" configuration so that the frigate can act as an offship ERT in case the Horizon gets into trouble, either against another party or against the Coalition. If used in its battle configuration it can support 15 crew at full capacity.
<details>
<summary>Screenshots</summary>

image
image
image
image
image
image
image
image

</details>

### CODE REVIEW

**Comment by nauticall** [comment]
!review

**Review by Eyeveri — CHANGES_REQUESTED** [review]
Hello! 

I went through the ship and it looks phenomenal. The code looks great to me as well, I didn't find any errors when I was testing it. 

However there are a few things that need to be adjusted. 

First - 
Ignore what I said about lockers, but I agree with dreamy. This is too powerful of a loadout for a third party ship. I would rather see this be in an ERT/Event type vessel. 
code into lockers please
CodeIntoLockers

Second - I did not see an IPC recharger anywhere. I might have missed it, but if there isn't one, please add one. 

Third - 
A lot of decals are missing under the doors. I did not see a single caution decal under the airlocks around the ship. Please add those. 
decals
NeedsDecals

Fourth - This is minor, but the surgical console in the medical bay is backwards so you can see the wiring. Please flip it to face north. 
FlipConsole

Fifth - The Security vendor needs to have it's access updated so that the crew can use it. Also, I forgot to check the suit modifying machines, but please make sure your ghost roles can use those too. 
needs access

Lastly, and this one isn't required, but consider putting the "gold" CE border around the wood tiles on the ship, like here for example. 
ConsiderWoodBorders

All in all, great work. Address these and I'll approve them afterwards.

File: maps/away/ships/xanu/xanu_frigate.dm [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	These same redundancies, designed for greater survivability in a peer-on-peer conflict, make the Rapier an expensive investment, one which has seen the Estoc survive in the navies of other Coalition states."
+	suffixes = list("ships/xanu/xanu_frigate.dmm")
+	sectors = list(ALL_COALITION_SECTORS)
+	spawn_weight_sector_dependent = list(SECTOR_LIBERTYS_CRADLE = 3)
+	spawn_weight = 1
+	ship_cost = 1
+	id = "xanu_frigate"
+	unit_test_groups = list(3)
+	shuttles_to_initialise = list(/datum/shuttle/autodock/overmap/xanu_fighter, /datum/shuttle/autodock/overmap/xanu_boarder)
+	template_flags = TEMPLATE_FLAG_SPAWN_GUARANTEED
```
Comment (DreamySkrell) [inline maps/away/ships/xanu/xanu_frigate.dm:14]: .

File: code/game/objects/structures/window.dm [hunk 2]
```diff
@@ -787,6 +787,12 @@
 /obj/structure/window_frame/update_icon()
 	SSicon_smooth.add_to_queue(src)
 
+/obj/structure/window/full/reinforced/ultra
```
Comment (DreamySkrell) [inline code/game/objects/structures/window.dm:790]: remove this definition, not really needed, and definitely for use ONLY by this xanu ship and not any other

File: code/game/objects/structures/full_window_frame.dm [hunk 3]
```diff
@@ -183,3 +182,6 @@
 
 /obj/structure/window_frame/shuttle/blue
 	color = "#6176a1"
+
+/obj/structure/window_frame/shuttle/coalition
```
Comment (DreamySkrell) [inline code/game/objects/structures/full_window_frame.dm:186]: remove, and use new system of `/obj/effect/map_effect/window_spawner/full/shuttle`, more on this below

File: code/modules/effects/map_effects/window_spawner.dm [hunk 4]
```diff
@@ -175,6 +175,28 @@
 	icon_state = "full_i_rwindow"
 	window_path = /obj/structure/window/full/reinforced/indestructible
 
+//Coalition window frames
+/obj/effect/map_effect/window_spawner/full/reinforced/coalition
```
Comment (DreamySkrell) [inline code/modules/effects/map_effects/window_spawner.dm:179]: these should all be a subtype of `/obj/effect/map_effect/window_spawner/full/shuttle`
icon states and grille/firedoor/etc subtypes are fine to keep for ease of use
`frame_path` should be normal window frames
`frame_color` set to `COLOR_COALITION`
`window_path` should be either normal reinforced or borosilicate, but I would prefer normal

**Comment by DreamySkrell** [comment]
image
these already are MRE rations crates, why map in even more MREs into them?

**Comment by DreamySkrell** [comment]
image
way too powerful ammo for the longbow - normal primers, lots of HE warheads, bunker busters
compare to gadpathur ship with only 4 low power primers and HE/AP warheads

**Comment by DreamySkrell** [comment]
image

**Comment by DreamySkrell** [comment]
image
do not route wires to EVERY window
the ship should

**Comment by DreamySkrell** [comment]
`/obj/machinery/shipsensors/strong` should be reserved for SCC ships, or survey or the like ships
use the normal sensors

**Comment by DreamySkrell** [comment]
image
remove the turrets - same thing as about the wires under windows

**Review by Eyeveri — APPROVED** [review]
All changes that were requested have been met. 

I think this ship is going to be a great addition, good work!

**Review by DreamySkrell — APPROVED** [review]

**Review by Eyeveri — APPROVED** [review]

**Review by alsoandanswer — APPROVED** [review]
Sprites look fine

**Pull request merged**